### PR TITLE
update dependency & readme file

### PR DESCRIPTION
**Pull request by ganganiSoN**
Changes:

- Replace Tslint with Eslint
- Update Reme.md
- Update all the packages with the latest version

Screenshot 2022-05-25 at 9 17 13 AM
Screenshot 2022-05-25 at 9 17 18 AM
Screenshot 2022-05-25 at 9 17 23 AM
Screenshot 2022-05-25 at 9 17 30 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,12 +28,21 @@
   "homepage": "https://github.com/pea3nut/short-night#readme",
   "devDependencies": {
     "@types/moveto": "^1.8.0",
-    "node-sass": "^4.11.0",
+    "@types/node": "^17.0.35",
```
Comment (pea3nut) [inline package.json]: Where is the `package-lock.json`? If you update the `package.json`, I think `package-lock.json` will be also updated accordingly and automatically.

File: package.json [hunk 2]
```diff
@@ -28,12 +28,21 @@
...
+    "@typescript-eslint/parser": "^5.25.0",
+    "eslint": "^8.16.0",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-airbnb-typescript": "^17.0.0",
+    "eslint-plugin": "^1.0.1",
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-jsx-a11y": "^6.5.1",
+    "eslint-plugin-react": "^7.30.0",
+    "eslint-plugin-react-hooks": "^4.5.0",
+    "node-sass": "^7.0.1",
```
Comment (pea3nut) [inline package.json]: Can we use `dart-sass` instead?

File: package.json [hunk 3]
```diff
@@ -28,12 +28,21 @@
...
-    "node-sass": "^4.11.0",
+    "@types/node": "^17.0.35",
+    "@typescript-eslint/eslint-plugin": "^5.25.0",
+    "@typescript-eslint/parser": "^5.25.0",
+    "eslint": "^8.16.0",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-airbnb-typescript": "^17.0.0",
+    "eslint-plugin": "^1.0.1",
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-jsx-a11y": "^6.5.1",
```
Comment (pea3nut) [inline package.json]: Why do we need JSX and React related dependency?

File: README.md [hunk 4]
```diff
@@ -16,9 +16,136 @@ An engine for draw a timeline graph:
...
+```
+
+# Usage
+
+```Javascript
+import * as Engine from 'short-night';
+```
+### For Creating Axis
+Create Axis Clasess same as below pattern. For AxisBody Extens `Engine.AxisBody`, AxisMilestone Extens `Engine.AxisMilestone`, AxisScale Extens `Engine.AxisScale`
+```
```
Comment (pea3nut) [inline README.md:34]: \`\`\`js

File: README.md [hunk 5]
```diff
@@ -16,9 +16,136 @@ An engine for draw a timeline graph:
...
 
+# Install
+
+NPM
+
+```
+npm install short-night
+```
+
+# Usage
```
Comment (pea3nut) [inline README.md:27]: In `README.md`, we should explain how to install and start development for short-night. Such build and setup related things. 

This section is more like a development manual, we should put them on our Wiki pages.

**Review by pea3nut — CHANGES_REQUESTED** [review]

**Comment by ganganiSoN** [comment]
- Developer Section
Screenshot 2022-06-03 at 2 00 21 AM
Screenshot 2022-06-03 at 2 00 27 AM

File: package.json (hunk 3 above)
Comment (ganganiSoN) [reply package.json]: Done.

File: package.json (hunk 1 above)
Comment (ganganiSoN) [reply package.json]: package-lock.json added.

File: package.json (hunk 2 above)
Comment (ganganiSoN) [reply package.json]: dart-sass is deprecated so, I used sass.

File: README.md (hunk 4 above)
Comment (ganganiSoN) [reply README.md:34]: done.

File: README.md (hunk 5 above)
Comment (ganganiSoN) [reply README.md:27]: Done.

Comment (pea3nut) [reply README.md:27]: How have you done it? I cannot find any improvement

File: README.md [hunk 6]
```diff
@@ -16,9 +16,185 @@ An engine for draw a timeline graph:
...
+
+Please make sure to update tests as appropriate.
+
+1. Fork the project on GitHub.
+
+2. Clone the project.
+
+3. perform the below command
+```sh
+npm link
```
Comment (pea3nut) [inline README.md:164]: Can we run `npm link` without any params?

File: package.json [hunk 7]
```diff
@@ -28,15 +28,22 @@
   "homepage": "https://github.com/pea3nut/short-night#readme",
   "devDependencies": {
     "@types/moveto": "^1.8.0",
-    "node-sass": "^4.11.0",
+    "@types/node": "^17.0.35",
```
Comment (pea3nut) [inline package.json]: Why do we need `@types/node`

File: package.json [hunk 8]
```diff
@@ -28,15 +28,22 @@
...
-    "typescript": "^3.0.0"
+    "typedoc": "^0.22.15",
+    "typescript": "^4.6.4"
   },
   "dependencies": {
     "html2canvas": "^1.0.0-rc.5",
     "moveto": "^1.8.0"
-  }
+  },
+  "main": "src"
```
Comment (pea3nut) [inline package.json:47]: I don't think `src/` should be the `main` of Package.json. Can you explain this further?

File: README.md (hunk 6 above)
Comment (ganganiSoN) [reply README.md:164]: yes, because the npm link performs the local linking process with a particular library.

**Pull request closed without merging**